### ISSUE TITLE
Auditors cannot view Instance Groups in UI

### ISSUE THREAD

**Issue by hjkatz** [ansible/awx#9036]
##### ISSUE TYPE
 - Bug Report

##### SUMMARY

Logged in as an admin I can see the following at `/instance_groups`

![image](https://user-images.githubusercontent.com/2234516/103909233-723e3b00-50d1-11eb-8888-c98d820f8c8c.png)

Logged in as an auditor I see the following at `/instance_groups`

![image](https://user-images.githubusercontent.com/2234516/103909260-7b2f0c80-50d1-11eb-84ba-9c9b44ad4531.png)

##### ENVIRONMENT
* AWX version: 16.0.0
* AWX install method: kubernetes

##### EXPECTED RESULTS

Auditors should have full system read, except for encrypted materials (depending on access).

##### ACTUAL RESULTS

Auditors cannot see Instance Groups.

**Comment by nixocio** [comment]
hey @hjkatz, thanks for filing this issue.

I got the latest version of devel branch and I could not recreate this one.

Organizations 

![image](https://user-images.githubusercontent.com/9053044/104242928-d5abce00-542d-11eb-8707-c6195be50f75.png)

Projects

![image](https://user-images.githubusercontent.com/9053044/104243005-ea886180-542d-11eb-94c4-bb3a0cd29370.png)

Users
![image](https://user-images.githubusercontent.com/9053044/104243356-6edae480-542e-11eb-8894-1c957fb9a441.png)


Instance Groups as Admin 

![image](https://user-images.githubusercontent.com/9053044/104243218-3dfaaf80-542e-11eb-9aee-d0d6dcc28d33.png)

Instance Groups as Auditor

![image](https://user-images.githubusercontent.com/9053044/104243435-8e720d00-542e-11eb-8b12-fc8a85d515c4.png)

Can you provide more details about your setup?

**Comment by hjkatz** [comment]
Thanks for trying this out. I noticed that you've given the User account a User Role of Auditor, whereas in my setup we have the User account granted a Team Role of Auditor (for the Organization). However, even testing this difference manually (by granting the Org Auditor Role directly to the test User), the problem persists.

As a note we are using AWX 16.0.0 and not the `devel` version.

Briefly our setup is the following:

* 1 shared Org (let's call it Org-A) that's different from the `default` Org (we've deleted the default org)
    * Inside Org-A are some number of Teams, each Team has some number of Users
* N Team Orgs (just an Org with the same name as a team, you can just make Org-B as a test)
    * Full Org-B Admin is granted as a Team Role from Org-A's Team (it's assumed that Users associated to the Team would then also be Admins)
* A Team in Org-A is granted Team Role of Auditor for Org-A
* The `admin` credentials are used to create the Instance Groups

In this setup our Users in Org-A's Auditor Team are unable to view Instance Groups in the UI.

**Comment by nixocio** [comment]
hey @hjkatz, I still not able to reproduce this one.

Please, can you rephrase this statement? Or provide more granular steps.

> N Team Orgs (just an Org with the same name as a team, you can just make Org-B as a test)

 >   Full Org-B Admin is granted as a Team Role from Org-A's Team (it's assumed that Users associated to the Team would then also be Admins)

Once you logged in as the auditor that can not see the instance groups.

1 - Share HTTP requests being made when attempting to see the instance groups screen. Something like:

![image](https://user-images.githubusercontent.com/9053044/104742846-f4a1ad00-5718-11eb-93ef-24b239f86758.png)

2 - Access the API endpoint`/api/v2/instance_groups/`, and verify that user can see the instance groups there. Something like:

![image](https://user-images.githubusercontent.com/9053044/104743488-c1135280-5719-11eb-9137-a659c1c80094.png)

**Comment by hjkatz** [comment]
Sure! Let me try and explain the setup we have for Orgs and Teams, because it's a bit complicated :)

## Minimal Org/Team/User Setup

First, I _think_ you should be able to reproduce with 2 Orgs and 1 Team. Let's give them some names and structure:

* First, delete the `default` Org, we don't use it.
* Create an Org called "Shared Org"
* Create an Org called "Personal Org"
* Create a Team in the Shared Org called Personal Team
* Add a User (let's call them Bob Jones) to the Personal Team (as a note, we use SAML for this step)
* Grant the Personal Team all Admin Roles (inventory, job, project, etc...) to the Personal Org
* Grant the User Bob Jones the Auditor Role over the Shared Org and the Personal Org
* Finally, as the AWX admin user, create Instance Groups

I _think_ in this minimal setup that the problem might appear. If not, then I'm happy to troubleshoot with you through other means.

## Networking Debugging

Here's some additional API, console, and networking information shown from the perspective of some Users:

### User: `admin` (the default AWX admin login)

Visiting `/api/v2/instance_groups`:

![image](https://user-images.githubusercontent.com/2234516/104744359-c7ee9500-571a-11eb-8745-e0b4aa1c989b.png)

![image](https://user-images.githubusercontent.com/2234516/104744419-db016500-571a-11eb-81e4-0f8058df4861.png)

```json
{
    "count": 2,
    "next": null,
    "previous": null,
    "results": [
        {
            "id": 1,
            "type": "instance_group",
            "url": "/api/v2/instance_groups/1/",
            "related": { ...  },
            "name": "tower",
            "created": "2020-12-11T20:45:15.956449Z",
            "modified": "2020-12-11T20:45:16.006823Z",
            "capacity": 20,
            "committed_capacity": 0,
            "consumed_capacity": 0,
            "percent_capacity_remaining": 100.0,
            "jobs_running": 0,
            "jobs_total": 1109,
            "instances": 1,
            "controller": null,
            "is_controller": false,
            "is_isolated": false,
            "is_containerized": false,
            "credential": null,
            "policy_instance_percentage": 100,
            "policy_instance_minimum": 0,
            "policy_instance_list": [],
            "pod_spec_override": "",
            "summary_fields": {
                "user_capabilities": {
                    "edit": true,
                    "delete": true
                }
            }
        },
        {
            "id": 2,
            "type": "instance_group",
            "url": "/api/v2/instance_groups/2/",
            "related": { ...  },
            "name": "Shared ansible repo Docker image",
            "created": "2020-12-17T16:46:30.596856Z",
            "modified": "2021-01-13T15:23:24.211597Z",
            "capacity": 0,
            "committed_capacity": 0,
            "consumed_capacity": 0,
            "percent_capacity_remaining": 0.0,
            "jobs_running": 0,
            "jobs_total": 79,
            "instances": 0,
            "controller": null,
            "is_controller": false,
            "is_isolated": false,
            "is_containerized": true,
            "credential": 1,
            "policy_instance_percentage": 0,
            "policy_instance_minimum": 0,
            "policy_instance_list": [],
            "pod_spec_override": "REDACTED",
            "summary_fields": {
                "credential": {
                    "id": 1,
                    "name": "AWX Kubernetes Service Account Bearer Token",
                    "description": "Kubernetes Service Account Bearer Token for default \"awx\" Service Account. Collected directly from Secret/awx-<...> in the Namespace. Token has been manually pulled and added to awx-config secrets.",
                    "kind": "kubernetes_bearer_token",
                    "cloud": false,
                    "kubernetes": true,
                    "credential_type_id": 16
                },
                "user_capabilities": {
                    "edit": true,
                    "delete": true
                }
            }
        }
    ]
}
```

### User: `auditor` (a user with the auditor role)

Visiting `/api/v2/instance_groups`:

![image](https://user-images.githubusercontent.com/2234516/104744741-3cc1cf00-571b-11eb-8d83-66d3af0dcb1a.png)

![image](https://user-images.githubusercontent.com/2234516/104744756-40edec80-571b-11eb-83d5-08961a2c277b.png)

```json
{
    "count": 0,
    "next": null,
    "previous": null,
    "results": []
}
```

Proof that the user is an auditor:

.... ok so when I got to this point I discovered something interesting. I'm going to leave the debugging information above, however, check this out!

At `/api/v2/me/` I see the following for the auditor user:

```json
{
    "count": 1,
    "next": null,
    "previous": null,
    "results": [
        {
            "id": 11,
            "type": "user",
            "url": "/api/v2/users/11/",
            "related": { ... },
            "summary_fields": {
                "user_capabilities": {
                    "edit": false,
                    "delete": false
                }
            },
            "created": "2020-12-15T23:44:06.919341Z",
            "username": "hkatz@squarespace.com",
            "first_name": "Harrison",
            "last_name": "Katz",
            "email": "hkatz@squarespace.com",
            "is_superuser": false,
            "is_system_auditor": false,
            "ldap_dn": "",
            "last_login": "2021-01-15T15:19:31.313005Z",
            "external_account": "social",
            "auth": [
                {
                    "provider": "saml",
                    "uid": "okta:Harrison Katz"
                }
            ]
        }
    ]
}
```

Check out that `is_system_auditor: false` part! However, the Team that I'm on _does_ have the auditor role:

From `/api/v2/teams/2/access_list/`

```json
...
{
            "id": 11,
            "type": "user",
            "url": "/api/v2/users/11/",
            "related": { ... },
            "summary_fields": {
                "direct_access": [
                    {
                        "role": {
                            "id": 19,
                            "name": "Member",
                            "description": "User is a member of the team",
                            "resource_name": "Compute",
                            "resource_type": "team",
                            "related": {
                                "team": "/api/v2/teams/2/"
                            },
                            "user_capabilities": {
                                "unattach": false
                            }
                        },
                        "descendant_roles": [
                            "member_role",
                            "read_role"
                        ]
                    },
                    {
                        "role": {
                            "id": 10,
                            "name": "Auditor",
                            "description": "Can view all aspects of the organization",
                            "team_id": 2,
                            "team_name": "Compute",
                            "team_organization_name": "Squarespace",
                            "resource_name": "Squarespace",
                            "resource_type": "organization",
                            "related": {
                                "organization": "/api/v2/organizations/1/"
                            },
                            "user_capabilities": {
                                "unattach": false
                            }
                        },
                        "descendant_roles": [
                            "member_role",
                            "read_role"
                        ]
                    },
                    {
                        "role": {
                            "id": 20,
                            "name": "Read",
                            "description": "May view settings for the team",
                            "team_id": 2,
                            "team_name": "Compute",
                            "team_organization_name": "Squarespace",
                            "resource_name": "Compute",
                            "resource_type": "team",
                            "related": {
                                "team": "/api/v2/teams/2/"
                            },
                            "user_capabilities": {
                                "unattach": false
                            }
                        },
                        "descendant_roles": [
                            "member_role",
                            "read_role"
                        ]
                    }
                ],
                "indirect_access": []
            },
            "created": "2020-12-15T23:44:06.919341Z",
            "username": "hkatz@squarespace.com",
            "first_name": "Harrison",
            "last_name": "Katz",
            "email": "hkatz@squarespace.com",
            "is_superuser": false,
            "is_system_auditor": false,
            "ldap_dn": "",
            "last_login": "2021-01-15T15:19:31.313005Z",
            "external_account": "social"
        },
...
```

Not sure if this is significant, but here's a bunch more information at least!

**Comment by hjkatz** [comment]
I think this is still a problem and I think it's related to the Team RBAC and permissions setup. Here's some related issues:
* https://github.com/ansible/awx/issues/9178
* https://github.com/ansible/awx/issues/9128

**Comment by jakemcdermott** [comment]
@hjkatz 

Would you say that the rest of this issue is covered by #9128? If so we'll close this one and focus our efforts on #9128

**Comment by hjkatz** [comment]
@jakemcdermott Yeah I think it's covered. In a broader sense I think @AlanCoding has a grander view for how to help fix the RBAC and permissions system in AWX. His comment here enlightened me greatly to the underlying structural constraints: https://github.com/ansible/awx/issues/9128#issuecomment-765649920
